bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/device/command/CachedCommand.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pjlinkdevice.internal.device.command;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.cache.ExpiringCache;
+
+/**
+ * CachedCommand wraps any command and caches its response for a configurable period of time.
+ *
+ * @author Nils Schnabel - Initial contribution
+ */
+@NonNullByDefault
+public class CachedCommand<ResponseType extends Response<?>> implements Command<ResponseType> {
+
+    private Command<ResponseType> cachedCommand;
+    private ExpiringCache<ResponseType> cache;
+
+    public CachedCommand(Command<ResponseType> cachedCommand) {
+        this(cachedCommand, 1000);
+    }
+
+    public CachedCommand(Command<ResponseType> cachedCommand, int expiry) {
+        this.cachedCommand = cachedCommand;
+        this.cache = new ExpiringCache<ResponseType>(expiry, () -> {
+            try {
+                return this.cachedCommand.execute();
+            } catch (ResponseException | IOException | AuthenticationException e) {
+                // wrap exception into RuntimeException to unwrap again later in CachedCommand.execute()
+                throw new RuntimeException(e);
```

Comment:
Do not throw raw exception types such as the `Exception` or the `RuntimeException`, use a specific exception and if none exists create your own.

Reply:
Done, there's a new CacheException now.